sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// AzureCLITokenProvider can be used to supply the AzureCLICredential with an alternate token provider
+type AzureCLITokenProvider func(ctx context.Context, resource string) ([]byte, error)
+
+// AzureCLICredentialOptions contains options used to configure the AzureCLICredential
+type AzureCLICredentialOptions struct {
+	TokenProvider AzureCLITokenProvider
+}
+
+// AzureCLICredential enables authentication to Azure Active Directory using the Azure CLI command "az account get-access-token".
+type AzureCLICredential struct {
+	tokenProvider AzureCLITokenProvider
+}
+
+// NewAzureCLICredential constructs a new AzureCLICredential with the details needed to authenticate against Azure Active Directory
+// options: configure the management of the requests sent to Azure Active Directory.
+func NewAzureCLICredential(options *AzureCLICredentialOptions) (*AzureCLICredential, error) {
+	if options == nil {
+		var o AzureCLICredentialOptions
+		o.TokenProvider = defaultTokenProvider()
+		options = &o
```

Comment:
Nit, you could condense this to the following.
```go
options = &AzureCLICredentialOptions{TokenProvider: defaultTokenProvider()}
```

Reply:
Done